barretenberg/cpp/src/barretenberg/sumcheck/sumcheck.hpp

```diff
@@ -7,6 +7,111 @@
 
 namespace bb {
 
+/*! \brief The implementation of the sum-check Prover for the statements of the form \f$\sum_{\vec \ell \in \{0,1\}^d}
+pow_{\beta}(\vec \ell) \cdot F \left(P_1(\vec \ell),\ldots, P_N(\vec \ell) \right)  = 0 \f$ for multi-linear polynomials
+\f$P_1, \ldots, P_N \f$.
+
+   \details
+ ## Notation and Setup
+
+ ### Obtaining Prover/Honk Polynomials
+ The Sumcheck is applied to a \p Flavor that in its turn specifies the information about the multi-variate polynomials
+\f$P_1, \ldots, P_N\f$. Namely, \ref prove "prove method" obtains \p full_polynomials by reference from \p Flavor 's
+\ref ProverPolynomials "prover polynomials". In particular, their number \f$N\f$ is specified by the \p Flavor.
+
+ ### Sumcheck Relation
+ Given multi-linear polynomials \f$ P_1,\ldots, P_N \in \mathbb{F}[X_0,\ldots, X_{d-1}] \f$ and a relation \f$ F \f$
```

Comment:
I think it's a bit confusing to refer to `F` as a polynomial, it's an identity, almost like a function applied on polynomials

Reply:
Yeah, called it a function. Thanks for pointing out!

Reply:
Actually, F has to be a polynomial in prover polynomials and hence a polynomial in X_0,..., X_{d-1}. Otherwise, round univariates would not be polynomials and many other things will break.